data/lua/core/stringx.lua

```diff
@@ -1,11 +1,22 @@
 --[========[Additional string support functions]========]
 print("Loading stringx module...")
 
+---Split a string on a separator, skipping separators that have been escaped
+---@param str string
+---@param sep? string
+---@param esc? string
+---@return string[]
 function stringx.escaped_split(str, sep, esc)
 	esc = esc or '\\'
 	return stringx.split(str, sep, {escape = esc, strip_spaces = true, remove_empty = true})
 end
 
+---Split a string on a separator, skipping separatorss that are enclosed in quotes
+---@param str string
+---@param sep? string
+---@param left? string
+---@param right? string
```

Comment:
Documentation is unclear about what `left` and `right` are. From the first line, I'd expect this to default to quotation marks.